### PR TITLE
Use challenge from params

### PR DESCRIPTION
**Pull request by akonior**

### CODE REVIEW

File: src/tests/webauthn_stub.ts [hunk 1]
```diff
@@ -1,96 +1,60 @@
...
       153, 92, 243, 186, 131, 29, 151, 99, 29, 0, 0, 0, 0,
     ];
+
+    const challengeBase64 = Buffer.from(challenge)
+      .toString('base64')
+      .replaceAll('+', '-')
+      .replaceAll('/', '_')
+      .replaceAll('=', '')
+      .split('')
+      .map(c => c.charCodeAt(0));
```
Comment (marekkirejczyk) [inline src/tests/webauthn_stub.ts:33]: Looks like a small method to extract.

File: src/app/account/webauthn_account_contract.ts [hunk 2]
```diff
@@ -73,7 +72,6 @@ class WebAuthnWitnessProvider implements AuthWitnessProvider {
     const witness = [
       ...signature.signatureRaw,
       ...signature.authenticatorData,
-      ...signature.challenge,
       signature.clientDataJson.length,
       ...signature.clientDataJson,
       ...new Uint8Array(CLIENT_DATA_JSON_MAX_LEN - signature.clientDataJson.length),
```
Comment (marekkirejczyk) [inline src/app/account/webauthn_account_contract.ts:77]: How about?

```ts
async  function createAuthWitness(message: Fr): Promise<AuthWitness> {
    const signature = await this.webAuthnInterface.sign(new Uint8Array(message.toBuffer()));
    return new AuthWitness(message, signature.toArray()));
}

class AuthInterface {
  ...
  function toArray(signature) : Uint8Array {
   ...
  }
...
}
```

File: src/app/account/webauthn_account_contract.ts [hunk 3]
```diff
@@ -73,7 +72,6 @@ class WebAuthnWitnessProvider implements AuthWitnessProvider {
     const witness = [
```
Comment (marekkirejczyk) [inline src/app/account/webauthn_account_contract.ts:72]: Remove TODO

File: src/app/components/webauthn/login.tsx [hunk 4]
```diff
@@ -2,11 +2,7 @@ import { convertBigIntToUint8Array, convertUint8ArrayToBigInt } from './convert.
 
 const SUPPORTED_CLIENT_DATA_JSON_LENGTHS = [114, 134, 243];
 
-export async function webAuthnLogin(
-  challenge: Uint8Array = new Uint8Array([
-    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-  ]),
-) {
+export async function webAuthnLogin(challenge: Uint8Array) {
```
Comment (marekkirejczyk) [inline src/app/components/webauthn/login.tsx:5]: Change tsx -> ts

File: src/contracts/webauthn_account_contract/src/utils.nr [hunk 5]
```diff
@@ -3,7 +3,7 @@ use dep::std;
 fn convert_field_to_array(f: Field) -> [u8; 32] {
     let mut res = [0; 32];
     let slice = f.to_be_bytes(32);
-    for i in 0..slice.len() {
+    for i in 0..32 {
```
Comment (marekkirejczyk) [inline src/contracts/webauthn_account_contract/src/utils.nr:6]: Extract constant?

**Review by marekkirejczyk — APPROVED** [review]

File: src/app/account/webauthn_account_contract.ts (hunk 2 above)
Comment (akonior) [reply src/app/account/webauthn_account_contract.ts]: Done

File: src/app/account/webauthn_account_contract.ts (hunk 3 above)
Comment (akonior) [reply src/app/account/webauthn_account_contract.ts]: Done

**Pull request merged**